fw/cache.c

```diff
@@ -1261,21 +1261,52 @@ tfw_cache_h2_copy_int(unsigned int *acc_len, unsigned long src,
 	return 0;
 }
 
+static int
+tfw_cache_copy_str_with_extra_quotes(TfwCacheEntry *ce, char **p, TdbVRec **trec,
+				     TfwStr *src, size_t *tot_len,
+				     bool need_extra_quotes)
+{
+#define ADD_ETAG_QUOTE(flag)                                            \
+do {                                                                    \
+        TfwStr quote = { .data = "\"", .len = 1, .flags = flag };       \
+        if (tfw_cache_h2_copy_str(&ce->hdr_len, p, trec, &quote,        \
+                                  tot_len))                             \
+                return -ENOMEM;                                         \
+} while(0)
+
+	if (need_extra_quotes)
+		ADD_ETAG_QUOTE(0);
```

Comment:
Why do we add first quote without `TFW_STR_VALUE` flag?

Reply:
Since we don't do it when we parse Etag value in `__parse_etag_or_if_nmatch` in http_parser.c. Here i implement the same behaviour

Reply:
It looks strange, part of value has "value" flag and part not. Do we have some places dependent of such behavior? I suggest: Find all places where it used and ensure that after changing behaviour we not broke something. If we don't broke anything, then add `TFW_STR_VALUE` for first quote and remove condition from `__set_etag`.

Reply:
No, you are wrong here! Etag value doesn't contains quotes! Quotes are only used to separate etag values, We save last quote value in etag only as a stop symbol for `tfw_strcmpspn`!